### PR TITLE
DM-39672: make it possible to force failures in mock pipelines on the command-line

### PR DESCRIPTION
**Pull request by TallJimbo**
## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/ctrl/mpexec/cli/opt/options.py [hunk 1]
```diff
@@ -440,6 +445,46 @@
...
+    callback=parse_mock_failure,
+    metavar="LABEL:EXCEPTION:WHERE",
+    default=None,
+    multiple=True,
+    help=unwrap(
+        """Specifications for tasks that should be configured to fail
+        when mocking execution.  This is a colon-separated 3-tuple,
+        where the first entry the task label, the second the
+        fully-qualified exception type (empty for ValueError, and the
+        third a string (usually quoted) of the form passed to --where,
```
Comment (andy-slac) [inline python/lsst/ctrl/mpexec/cli/opt/options.py:482]: "usually quoted" may be a bit confusing. I guess quoting is needed for shell only, not for the Python code that parses it?

**Review by andy-slac — APPROVED** [review]
Looks good

File: python/lsst/ctrl/mpexec/cli/opt/options.py [hunk 2]
```diff
@@ -440,6 +445,46 @@
...
+) -> Mapping[str, tuple[str, type[Exception] | None]]:
+    """Parse the --mock-failure option values into the mapping accepted by
+    `lsst.pipe.base.tests.mocks.mock_task_defs`.
+    """
+    result: dict[str, tuple[str, type[Exception] | None]] = {}
+    if value is None:
+        return result
+    for entry in value:
+        try:
+            task_label, error_type_name, where = entry.split(":")
```
Comment (andy-slac) [inline python/lsst/ctrl/mpexec/cli/opt/options.py:460]: One more comment - what if `where` also contains colon (e.g. for slicing)?

File: python/lsst/ctrl/mpexec/cli/opt/options.py (hunk 1 above)
Comment (TallJimbo) [reply python/lsst/ctrl/mpexec/cli/opt/options.py]: Yes, just for the shell.  Reworded to that effect.

File: python/lsst/ctrl/mpexec/cli/opt/options.py (hunk 2 above)
Comment (TallJimbo) [reply python/lsst/ctrl/mpexec/cli/opt/options.py]: Good catch.  I've updated this to
```
task_label, error_type_name, *where_terms = entry.split(":")
where = ":".join(where_terms)
```
to guard against that.

Comment (andy-slac) [reply python/lsst/ctrl/mpexec/cli/opt/options.py]: `entry.split(":", 2)` is probably faster 🙂

Comment (TallJimbo) [reply python/lsst/ctrl/mpexec/cli/opt/options.py]: Neat, I didn't know about that argument.

**Pull request merged**